esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -422,15 +425,38 @@ def main(cfg):
     """Perform climwip weighting method."""
     models, observations = read_metadata(cfg)
 
-    variable_groups = models.keys()
+    variable_groups_independence = [
+        key for key, value in cfg['independence_contributions'].items()
+        if value > 0
+    ]
+    variable_groups_performance = [
+        key for key, value in cfg['performance_contributions'].items()
+        if value > 0
+    ]
+    assert len(variable_groups_independence) > 0
+    assert len(variable_groups_performance) > 0
 
     model_ancestors = []
     obs_ancestors = []
 
     performances = {}
     independences = {}
 
-    for variable_group in variable_groups:
+    for variable_group in variable_groups_independence:
```

Comment:
I wonder how much overhead it is to calculate performance/independence for those variable groups that you'll eventually not include in the final weights. If it's not that much, we wouldn't have to break up this loop.

Reply:
I personally find this cleaner, in particular in anticipation of the perfect model test, which will break symmetry between performance and independence even more. In addition it might be nice to be able to have the possibility of using independence variable groups for which no observations are available which should work like this I suspect (otherwise it will at least make it easier if we ever need it)